sphinx_selective_exclude/eager_only.py

```diff
@@ -31,9 +31,13 @@ def run(self, *args):
         env = self.state.document.settings.env
         env.app.builder.tags.add('TRUE')
         #print(repr(self.arguments[0]))
-        if not env.app.builder.tags.eval_condition(self.arguments[0]):
-            return []
-
+        try:
+            if not env.app.builder.tags.eval_condition(self.arguments[0]):
+                return []
+        except Exception as err:
+            logger = logging.getLogger(__name__)
+            logger.warning(('exception while evaluating only directive expression: %s'), err,
+                            location=(self.env.docname, self.lineno))
```

Comment:
are we sure we just want to log a warning and continue building? There probably wont be any html/latex generated for this directive due to this syntax error. And not every rst-write cares too much about terminal warnings. Two suggestions:

- or 1, we still fail the sphinx-build but give more details on where the syntax error is located
- or 2, we add some warning in the generated html/latex as well stating there was a syntax error,

for option 1, we could derive a specific Exception class where we added more details. For an example see https://github.com/melexis/sphinx-traceability-extension/blob/master/mlx/traceability_exception.py#L38-L60

Reply:
Stein,
 
I intended to raise. So I corrected this already.

For this exception class. I do not know whether there is much added value at this point.
This sphinx_selective_exclude is only 1 file so it is still clean like it is added now.

Usually I add this class the second time I add the same code.

Piet